### PR TITLE
Add TakeANumber

### PR DESCRIPTION
**Pull request by angelikatyborska**
As described in: https://github.com/exercism/v3/blob/6576904cbe3ddae26b23a418da8e012162980c4a/languages/elixir/exercises/concept/take-a-number/.meta/design.md#L33

### CODE REVIEW

File: test/elixir_analyzer/exercise_test/take_a_number_test.exs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  describe "forbids abstractions" do
+    test_exercise_analysis "detects Agent",
+      comments_include: [Constants.take_a_number_do_not_use_abstractions()],
+      status: :disapprove do
+      defmodule TakeANumber do
+        use Agent
+
+        def start() do
+          {:ok, agent_pid} = Agent.start_link(fn -> 0 end)
```
Comment (angelikatyborska) [inline test/elixir_analyzer/exercise_test/take_a_number_test.exs:42]: I know this is a really bizarre solution, but it passes all the tests...

**Comment by neenjaw** [comment]
I've been holding off on this because I thought I would be able to finish #32 sooner.  But with other things, this has gotten delayed.  I have opened up #34 which introduces some `lib` folder organizational changes (I don't think it will change much in here other than the location of the module) but it should probably be merged first, then this rebased on top of that.

**Review by neenjaw — APPROVED** [review]

**Comment by neenjaw** [comment]
On second thought, I'm going to merge this and rebase the refactor

**Comment by angelikatyborska** [comment]
Thanks!

**Pull request merged**